src/Relation.php

```diff
@@ -62,7 +62,7 @@ public static function getDefaultCandidateKey(Model $source)
      */
     public static function getDefaultForeignKey(Model $source)
     {
-        $tableName = $source->getTableName();
+        $tableName = $source->getTableAlias();
```

Comment:
I'm not sure this is sensible. This is not about a relation path. It's about a database column name.

If I have the following:

Table **rack** (pk: id)
Table **host** (pk: id, fk: rack_id)

I wouldn't expect that If i set an alias for **rack** that it mangles my relations because the default fk is now `$alias_id` instead of `rack_id`.

Though, if my real table name is **$prefix_rack**, I would still have a **rack_id** not a **$prefix_rack_id**, or would I? I'm not sure? @lippserd Your opinion please.

Reply:
> Though, if my real table name is **$prefix_rack**, I would still have a **rack_id** not a **$prefix_rack_id**, or would I?

As far as I know, you always have to define it explicitly in the relations as a foreign/candidate key, when the foreign key does not match this schema `tablename_id`.

Reply:
Yes! But then this change is redundant. Because if my table name includes a prefix, I deviate from the schema. Or, if my foreign key name is not `$prefix_rack_id` I also deviate.

**Fine:**
Table: `rack`
FK: `rack_id`

**Fine:**
Table: `$prefix_rack`
FK: `$prefix_rack_id`

**Wrong:**
Table: `$prefix_rack`
FK: `rack_id`